tests/src/Kernel/VocabularyReferenceFormatterTest.php

```diff
@@ -0,0 +1,400 @@
+
+    // Verify the items in the first list.
+    $items = $lists->eq(0)->filter('ul li');
+    $this->assertCount(3, $items);
+    $this->assertEquals('Italy', $items->eq(0)->html());
+    $this->assertEquals('Lithuania', $items->eq(1)->html());
+    $this->assertEquals('Belgium', $items->eq(2)->html());
+    // And in the second.
+    $items = $lists->eq(1)->filter('ul li');
+    $this->assertCount(1, $items);
+    $this->assertEquals('Italy', $items->html());
+    // Amd finally the third.
+    $items = $lists->eq(2)->filter('ul li');
+    $this->assertCount(1, $items);
+    $this->assertEquals('Referenced', $items->html());
+
+    // Set the formatter to link to the entity.
+    \Drupal::service('entity_display.repository')
+      ->getViewDisplay('entity_test', 'entity_test')
+      ->setComponent('field_test', [
+        'type' => 'open_vocabulary_reference_label',
+        'settings' => [
+          'link' => TRUE,
+        ],
+      ])
+      ->save();
+
+    $build = $entity->get('field_test')->view('default');
+    $html = (string) \Drupal::service('renderer')->renderRoot($build);
+    $crawler = new Crawler($html);
+
+    // Verify that the labels are linked to the entity path.
+    $lists = $crawler->filter('div.field__item div.item-list');
+    $links = $lists->eq(0)->filter('ul li a');
+    $this->assertEquals($this->countries['Italy']->toUrl()->toString(), $links->eq(0)->attr('href'));
+    $this->assertEquals('Italy', $links->eq(0)->html());
+    $this->assertEquals($this->countries['Lithuania']->toUrl()->toString(), $links->eq(1)->attr('href'));
+    $this->assertEquals('Lithuania', $links->eq(1)->html());
+    $this->assertEquals($this->countries['Belgium']->toUrl()->toString(), $links->eq(2)->attr('href'));
+    $this->assertEquals('Belgium', $links->eq(2)->html());
+    // Verify also the second list.
+    $links = $lists->eq(1)->filter('ul li a');
+    $this->assertEquals($this->countries['Italy']->toUrl()->toString(), $links->eq(0)->attr('href'));
+    $this->assertEquals('Italy', $links->eq(0)->html());
+    // And the third.
+    $links = $lists->eq(2)->filter('ul li a');
+    $this->assertEquals($referenced->toUrl()->toString(), $links->eq(0)->attr('href'));
+    $this->assertEquals('Referenced', $links->eq(0)->html());
+
+    // Set Italian as the current language.
+    \Drupal::service('language.default')->set(ConfigurableLanguage::load('it'));
+    \Drupal::languageManager()->reset();
+
+    $build = $entity->get('field_test')->view('default');
+    $html = (string) \Drupal::service('renderer')->renderRoot($build);
+    $crawler = new Crawler($html);
+
+    // Verify that the labels are translated when a translation is available.
+    $lists = $crawler->filter('div.field__item div.item-list');
+    $links = $lists->eq(0)->filter('ul li a');
+    $this->assertEquals('Italia', $links->eq(0)->html());
+    $this->assertEquals('Lituania', $links->eq(1)->html());
+    $this->assertEquals('Belgium', $links->eq(2)->html());
```

Comment:
Do we test the change of ordering of the multiple values withing the same association? I could not find it.

Reply:
The field item list is tested for order, and here I create Belgium before Lithuania bu reference Lithuania then Belgium to show that order of the items is maintained.